src/Adapter/Country/AbstractCountryHandler.php

```diff
@@ -29,14 +29,29 @@
 namespace PrestaShop\PrestaShop\Adapter\Country;
 
 use Country;
+use PrestaShop\PrestaShop\Adapter\Country\Repository\CountryRepository;
+use PrestaShop\PrestaShop\Core\Domain\Country\Command\EditCountryCommand;
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CannotEditCountryException;
 use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
 use PrestaShopException;
 
 /**
  * Abstract country handler
  */
 class AbstractCountryHandler
```

Comment:
I know its not you that created this, but have in mind that having abstract handlers for cases like this is not the best practice. 
It only has the validation, so instead there should be some dedicated validator that could be used anywhere (not only in the handlers).

Anyway, as there is already the validateFields method, I think you can reuse it as it is now :man_shrugging:

Reply:
hmm i added this but i m trying to keep same logic as in other cases. I see that most of the other cases has validate function like this in abstract. I guess i can move it to validator but having different logic then in other cases is confusing no?

Reply:
Yeah, it was previous practice that we used, but it wasn't the best and we have a new one starting from product page.
So if it was you that introduced this abstract handler and its not yet in production, then I think you can inspire from product and combination validators and do the same for country :thinking: (especially considering the fact you are out of tasks for now :smile:)